src/custom/components/AccountDetails/Transaction.tsx

```diff
@@ -51,45 +63,255 @@ function determinePillColour(status: ActivityStatus, type: ActivityType) {
   }
 }
 
-function getActivitySummary(params: {
-  id: string
-  activityData: ReturnType<typeof useActivityDescriptors>
-  suffix?: string
-}): string | null {
-  const { id, activityData, suffix } = params
-
-  if (!activityData) return null
+const Wrapper = styled.div`
+  display: flex;
+  flex-flow: column wrap;
+  width: 100%;
+  border-bottom: 1px solid #d9e8ef;
+`
 
-  const { summary } = activityData
+const IconType = styled.div`
+  height: 36px;
+  width: 36px;
+  position: relative;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  ${({ theme }) => theme.mediaWidth.upToMedium`
+      display: none;
+  `};
+  &::before {
+    content: '';
+    display: block;
+    background: ${({ color }) => color};
+    position: absolute;
+    top: 0;
+    left: 0;
+    height: inherit;
+    width: inherit;
+    border-radius: 36px;
+    opacity: 0.1;
+  }
+  svg {
+    display: flex;
+    margin: auto;
+  }
+  svg > path {
+    width: 100%;
+    height: 100%;
+    object-fit: contain;
+    margin: auto;
+    display: block;
+    fill: ${({ color }) => color};
+  }
+  // Loader
+  ${StyledSVG} {
+    > path {
+      fill: transparent;
+      stroke: ${({ color }) => color};
+    }
+  }
+`
 
-  let baseSummary = summary
+const Summary = styled.div`
+  display: flex;
+  flex-flow: column wrap;
+  color: ${({ theme }) => theme.text2};
+  > b {
+    color: inherit;
+    font-weight: 500;
+    line-height: 1;
+    font-size: 15px;
+    margin: 0 0 5px;
+  }
+  > div {
+    display: flex;
+    flex-flow: column wrap;
+    width: 100%;
+    opacity: 0.75;
+    font-size: 13px;
+  }
+  > div > span {
+    display: grid;
+    color: inherit;
+    grid-template-rows: 1fr;
+    grid-template-columns: 100px 1fr;
+    ${({ theme }) => theme.mediaWidth.upToMedium`
+      grid-template-columns: 1fr;
+      margin: 0 16px 8px 0;
+    `};
+  }
+  > div > span > b,
+  > div > span > i {
+    position: relative;
+    font-size: inherit;
+    font-weight: 500;
+    margin: 0;
+    color: inherit;
+    display: flex;
+    align-items: center;
+    font-style: normal;
+  }
+  > div > span > i {
```

Comment:
you really dont like cascading do you 😂 why not just write this all indented as part of the first `div` and or `div > span`? makes it a bit easier to read imo but all good.

Reply:
I think the reverse is true: Find this easier to read without cascading/nesting.

Reply:
Perhaps a better approach is to cut a line at the 3 element in a selector, and decide to make it a component. I can certainly see your argument in terms of these kind of multi nested selectors, potentially introducing errors at some point. Will re-factor.